backend/ticketvise/mail/retrieve.py

```diff
@@ -0,0 +1,204 @@
+from ticketvise.models.notification import Notification
+import email
+import imaplib
+import logging
+import poplib
+import re
+import ssl
+from uuid import UUID
+from django.core.mail import send_mail
+
+from django.db import transaction
+from django.db.models import Q
+from email_reply_parser import EmailReplyParser
+
+from ticketvise.models.comment import Comment
+from ticketvise.models.inbox import Inbox
+from ticketvise.models.ticket import Ticket
+from ticketvise.models.user import User, UserInbox
+
+
+def is_autogenerated(message):
+    # https://www.arp242.net/autoreply.html
+    return any([
+        message.get(
+            "Auto-Submitted") and message.get("Auto-Submitted").strip().lower() != "no",
+        message.get("X-Auto-Response-Suppress") and message.get(
+            "X-Auto-Response-Suppress").strip().lower() in ("dr", "autoreply", "all"),
+        message.get("Precedence") and message.get(
+            "Precedence").strip().lower() in ("bulk", "auto_reply", "list"),
+        message.get(
+            "X-Autoreply") and message.get("X-Autoreply").strip().lower() == "yes",
+        message.get("List-Id"),
+        message.get("List-Unsubscribe"),
+        message.get("Feedback-ID"),
+        re.match(message.get("From").lower(), "^no.?reply@"),
+        "mailer-daemon" in message.get("From").lower(),
+        "Mail Delivery Subsystem" in message.get("From").lower(),
+        message.get("X-Mailer") and message.get("X-Mailer").strip(
+        ).lower() == "Microsoft CDO for Windows 2000".lower()
+    ])
+
+
+def parse_message_id(raw_message_id):
+    parts = raw_message_id.split("@")
+    if len(parts) > 1:
+        try:
+            return UUID(parts[0].replace("<", "").strip().lower())
+        except Exception:
+            return None
+
+    return None
+
+
+def get_ticket_by_email_message_id(message):
+    references = message.get("References")
+    if not references:
+        return None
+
+    message_id = parse_message_id(references.split()[0])
+    if not message_id:
+        return None
+
+    notification = Notification.objects.filter(email_message_id=message_id).select_subclasses().first()
+    if not notification:
+        return None
+
+    return notification.ticket
+
+
+@transaction.atomic
+def submit_email_ticket(message: email.message.EmailMessage):
+    if is_autogenerated(message):
+        logging.warning(f"Ignored autogenerated message")
+        return
+
+    email_from = message.get("From")
+    subject = message.get("Subject")
```

Comment:
kan een subject empty zijn? zo ja, moet er dan een default "no subject" in komen

Reply:
.get() kent als het goed is een default value, zo hoef je niet een aparte check te doen?

Reply:
Ja idd, fixed again.